### PR TITLE
Add Public API Analyzer to the Shipping products

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

A better way to track the public api we are shipping. 

right now just enabled for Maui.Core

### Issues Fixed

None

Fixes #

### CODE REVIEW

File: src/Core/src/Core.csproj [hunk 1]
```diff
@@ -21,4 +21,15 @@
   <ItemGroup>
     <ProjectReference Include="..\..\Essentials\src\Essentials.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" PrivateAssets="All" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <NoWarn>$(NoWarn);RS0041</NoWarn>
```
Comment (mattleibow) [inline src/Core/src/Core.csproj:29]: what is this warning?

Comment (rmarinho) [reply src/Core/src/Core.csproj:29]: lol not sure i copied from blazor :P

Comment (mattleibow) [reply src/Core/src/Core.csproj:29]: @SteveSandersonMS can you tell us?

Comment (SteveSandersonMS) [reply src/Core/src/Core.csproj:29]: It’s the “oblivious reference types” warning (https://github.com/dotnet/roslyn-analyzers/issues/3793)

I’d recommend removing this suppression from your project and only putting it back if you really need it.

**Review by mattleibow — APPROVED** [review]
Added the PublicAPI files to:
 - Controls.Core
 - Controls.Xaml
 - Core
 - Essentials

**Pull request merged**